src/view/RoomsView.java

```diff
@@ -72,6 +77,57 @@ public void keyPressed(KeyEvent e) {}
           public void keyReleased(KeyEvent e) {}
         });
 
+    emailTextField.addKeyListener(
+        new KeyListener() {
+          @Override
+          public void keyTyped(KeyEvent e) {
+            RoomsState currentState = viewModel.getState();
+            String currentEmail = currentState.getEmail();
+            String email = "";
+            if (currentEmail != null) {
+              email = currentEmail;
+            }
+            currentState.setEmail(email + e.getKeyChar());
+            viewModel.firePropertyChanged();
+          }
+
+          @Override
+          public void keyPressed(KeyEvent e) {}
+
+          @Override
+          public void keyReleased(KeyEvent e) {}
+        });
+
+    // This is to resolve a crazy frontend issue
+    createRoomTextField.addKeyListener(
+        new KeyListener() {
+          @Override
+          public void keyTyped(KeyEvent e) {
+            // Handle in keyPressed
+          }
+
+          @Override
+          public void keyPressed(KeyEvent e) {
+            RoomsState currentState = viewModel.getState();
+            String currentCreateRoom = currentState.getCreateRoom();
+            String createRoom = currentCreateRoom != null ? currentCreateRoom : "";
+
+            if (e.getKeyCode() == KeyEvent.VK_BACK_SPACE || e.getKeyCode() == KeyEvent.VK_DELETE) {
```

Comment:
:cry: If we really need this, I'd rather use a document listener on the textinputfield (with a comment indicating that we need to handle backspaces):


```java
createRoomTextField.getDocument().addDocumentListener(
  new DocumentListener() {
                @Override
                public void insertUpdate(DocumentEvent e) {
                    update();
                }

                @Override
                public void removeUpdate(DocumentEvent e) {
                    update();
                }

                @Override
                public void changedUpdate(DocumentEvent e) {
                    update();
                }

                protected void update() {
                    String text = createRoomTextField.getText();

                }
})

Reply:
I'll refactor this in a future pr that finalizes `RoomsView` cosmetics.